### ISSUE TITLE
DTRX and docker saved images

### ISSUE THREAD

**Issue by erhan-** [dtrx-py/dtrx#13]
I have a wrapper for dtrx and use it in a batch process.

When docker images are saved like in the documentation and dtrx is run on the exported tar.gz, it extracts everything and somewhere runs cpio on it.  During this command it gets stuck and shows error about malformed content. I have to kill the cpio command at that moment so that my batch process continues. Has anyone extracted a docker image with dtrx before?

**Comment by noahp** [comment]
I'll take a look!

**Comment by erhan-** [comment]
After much digging I found a file where this occurs:
When I run dtrx -rn on this. It silently fails until you kill the cpio subprocess it runs. Then you will see errors like:

```
cpio: Malformed number |#x;
cpio: Malformed number |#x;`
cpio: Malformed number x|#x;`
cpio: Malformed number |#x;`
cpio: Malformed number #x;`
cpio: Malformed number #x;`
cpio: Malformed number x;`
```

```
!$ file cpio
cpio: ELF 32-bit MSB executable, PowerPC or cisco 4500, version 1 (SYSV), dynamically linked, interpreter /lib/ld.so.1, for GNU/Linux 2.6.4, stripped

```

**Comment by noahp** [comment]
AH! the problem is due to cpio not doing magic number verification before attempting extraction, and it crashes/hangs (depending on the particular `cpio` binary it's attempting to extract).

Reproducing is quite easy:

```bash
# this image contains the cpio binary that causes the extraction to hang
❯ docker pull alpine:3.13.6
❯ docker image save alpine:3.13.6 -o alpine.tar.gz
❯ dtrx -rn alpine.tar.gz
```

The fundamental command that hangs is:

```bash
# note: this is the extracted image from above
❯ cpio -i --make-directories --quiet --no-absolute-filenames --file alpine/f7055e235a8665ac2ae79f29bd773c7a40b409e9c5d71905fb6bcb6458d9b66a/layer/usr/bin/cpio
```

Your fix looks good! I've put a PR up at #15.
